src/daft-core/src/array/ops/cast.rs

```diff
@@ -390,6 +391,81 @@ impl TimestampArray {
 }
 
 impl TimeArray {
+    pub fn hour(&self) -> DaftResult<UInt32Array> {
+        let physical = self.physical.as_arrow();
+        let tu = match self.data_type() {
+            DataType::Time(time_unit) => time_unit.to_arrow(),
+            _ => unreachable!("TimeArray must have Time datatype"),
+        };
+
+        let date_arrow = physical
+            .iter()
+            .map(|ts| {
+                ts.map(|ts| {
+                    let naive_time =
+                        arrow2::temporal_conversions::timestamp_to_datetime(*ts, tu, &chrono::Utc)
+                            .time();
+                    naive_time.hour() as u32
+                })
+            })
+            .collect::<Vec<_>>();
+
+        UInt32Array::new(
+            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
+            Box::new(PrimitiveArray::from(date_arrow)),
+        )
+    }
+
+    pub fn minute(&self) -> DaftResult<UInt32Array> {
+        let physical = self.physical.as_arrow();
+        let tu = match self.data_type() {
+            DataType::Time(time_unit) => time_unit.to_arrow(),
+            _ => unreachable!("TimeArray must have Time datatype"),
+        };
+
+        let date_arrow = physical
+            .iter()
+            .map(|ts| {
+                ts.map(|ts| {
+                    let naive_time =
+                        arrow2::temporal_conversions::timestamp_to_datetime(*ts, tu, &chrono::Utc)
+                            .time();
+                    naive_time.minute() as u32
+                })
+            })
+            .collect::<Vec<_>>();
+
+        UInt32Array::new(
+            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
+            Box::new(PrimitiveArray::from(date_arrow)),
+        )
+    }
+
+    pub fn second(&self) -> DaftResult<UInt32Array> {
+        let physical = self.physical.as_arrow();
+        let tu = match self.data_type() {
+            DataType::Time(time_unit) => time_unit.to_arrow(),
+            _ => unreachable!("TimeArray must have Time datatype"),
+        };
+
+        let date_arrow = physical
+            .iter()
+            .map(|ts| {
+                ts.map(|ts| {
+                    let naive_time =
+                        arrow2::temporal_conversions::timestamp_to_datetime(*ts, tu, &chrono::Utc)
+                            .time();
+                    naive_time.second() as u32
+                })
+            })
+            .collect::<Vec<_>>();
+
+        UInt32Array::new(
+            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
+            Box::new(PrimitiveArray::from(date_arrow)),
+        )
+    }
+
```

Comment:
`cast.rs` wouldn't be the right place to put this, maybe you could put it in `array/ops/date.rs` and rename it to `time.rs` since that would encompass both date, timestamp, and time